### DISCUSSION TITLE
Drop into editor API support

### DISCUSSION THREAD

**Discussion by celynw** [James-Yu/LaTeX-Workshop#3417] [Q&A]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you can not see the requested feature. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before requesting a feature.

You visited the wiki?: Yes

### If your requested feature is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: Yes

## Is your requested feature  related to a problem? Please describe. [Required]

No

## To Reproduce  [Required]

1. Drag and drop an image from the explorer pane into an editor pane.
1. Hold shift (as suggested) to insert that file into the TeX.
1. It will drop the relative path of that image file into the body.

## Describe the solution you'd like  [Required]

It would be pretty cool if it also inserted the rest of the text in order to include that image, for example:

```tex
\begin{figure}[h]\centering
	\includegraphics[width=0.8\linewidth]{plot1.png}%
	\caption{\label{fig}Caption}
\end{figure}
```

This [API was mentioned](https://code.visualstudio.com/updates/v1_70#_drop-into-editor-api) on the release notes for VSCode version 1.70.0, so I think it's quite new. Here's an example for a markdown context.

![image](https://code.visualstudio.com/assets/updates/1_70/api-drop.gif)

There's also a sample of how to implement it [here](https://github.com/microsoft/vscode-extension-samples/tree/main/drop-on-document).

However, I'm aware there are many, many ways to do this and everybody has their preferences. I imagine this could be user-configurable, but that might also be a difficult task to solve.
So if this is out-of-scope it's no problem. I just thought I should bring up the possibility.
As it stands, dragging in a file to paste in the path is already useful, and doesn't irritate anybody,

I also imagine this could be helpful for other file types, but I'm not sure what they are at the moment 😅 

## Logs [Required]

N/A

## Screenshots

![image](https://user-images.githubusercontent.com/3299161/183045764-b940a889-2651-452f-b0bb-6640c32d67c0.png)

![image](https://user-images.githubusercontent.com/3299161/183045850-fa49aefb-6e4a-45b1-8d6c-9c0cc0e94626.png)

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: Windows 11 (likely all versions)
 - VS Code version: 1.70.0 (likely not before this version!)
 - Extension version: 8.28.0
 - TeX distribution version: TeX Live 2019/2022

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

Yes/No

If the answer is Yes, please write which one you are using. Write the versions of the remote extension.

- Remote SSH
- Remote Extension Version: 0.84.0

### Additional context

N/A

**Comment by James-Yu** [comment]
`LaTeX Utensils` previously has this feature, but the extension was no longer maintained. The feature can be more complex than it looks like, and unlikely maintained by the core dev of LW. I'm closing this issue as not planned, and friendly discussions are welcomed.

**Reply by leoleoasd** [reply]
Hi @James-Yu, I'm the new maintainer of LaTeX-Utilities. It's actively maintained now.

This is an exciting feature, but I'm not sure if LaTeX-Utilities have this feature now (we do support pasting image though), I'll check and add it.

**Reply by Enough7** [reply]
Any new info about this feature? @James-Yu  @leoleoasd 
I am willing to help out

**Reply by leoleoasd** [reply]
@Enough7  It should support pasting image. dropping is currently not supported, feel free to open a PR and I'll be happy to answer your questions.
